torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -741,13 +491,212 @@ Stmt* LoopNest::lowerToStmt(Tensor* t) {
   return body;
 }
 
+class FunctionInliner : public IRMutator {
+ public:
+  FunctionInliner(Store* producer)
+      : buf_(producer->buf()), producer_(producer) {
+    for (auto* i : producer->indices()) {
+      const Var* index_var = dynamic_cast<const Var*>(i);
+      if (index_var == nullptr) {
+        throw std::logic_error("cannot inline Buf with compound indices");
+      }
+      index_vars_.insert(index_var);
+    }
+  }
+
+ private:
+  // For the target function, insert the caller/callee pair into the replacement
+  // mapping.
+  const Expr* mutate(const FunctionCall* v) override {
+    Function* func = v->tensor()->function();
+    const Buf* buf = v->tensor()->buf();
+    if (buf != buf_) {
+      return IRMutator::mutate(v);
+    }
+
+    // TODO: Support multiple-output functions
+    if (func->func_vars().size() != 1) {
+      throw unimplemented_lowering();
+    }
+
+    std::vector<const Var*> index_vars;
+    assert(buf->ndim() == func->args.size());
```

Comment:
() seem to be missing after 'func->args', wouldn't it miscompile?

Reply:
Huh, it should fail yeah, but it didn't. Exceptions are turned off maybe

Reply:
Nit: I think guidelines recommend to use `TORCH_INTERNAL_ASSERT` for asserts.

Reply:
Cool, I'll use that then.